### PR TITLE
Issue 1

### LINKED ISSUE #1: As a developer, I want doxygen formatted documentation of the main.c source file, so that functionality is more easily understood

### PR DESCRIPTION
**Pull request by fosterj14**
There were a few spots I was unsure of so I marked them with "Unsure".

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -152,6 +214,12 @@ GLuint gen_text_buffer(float x, float y, float n, char *text)
     return gen_faces(4, length, data);
 }
 
+/**
+Used to draw chunks and items in the game world
+\param[in] attrib: Attrib struct that contains information on what will be drawn.
```
Comment (lukemoore495) [inline src/main.c]: This sounds like attrib is going to be modified inside the function. If that's true, then I think the param is technically supposed to be "param[in,out]" to indicate that it has side effects.

File: src/main.c [hunk 2]
```diff
@@ -171,6 +239,12 @@ void draw_triangles_3d_ao(Attrib *attrib, GLuint buffer, int count)
     glBindBuffer(GL_ARRAY_BUFFER, 0);
 }
 
+/**
+Used to draw text that displays on signs in the world.
+\param[in] attrib: Attrib struct that contains information on what will be drawn.
```
Comment (lukemoore495) [inline src/main.c]: Assuming my above comment is true, change all attrib to be "param[in,out]"

File: src/main.c [hunk 3]
```diff
@@ -265,21 +385,41 @@ void draw_sign(Attrib *attrib, GLuint buffer, int length)
...
+*/
 void draw_plant(Attrib *attrib, GLuint buffer)
 {
     draw_item(attrib, buffer, 24);
 }
 
+/**
+Handles drawing all players that appear in the world.
+\param[in] attrib: Attrib struct that contains information on what will be drawn.
+\param[in] buffer: Buffer that will be used for the drawing.
```
Comment (lukemoore495) [inline src/main.c:411]: Should be "player", though it does appear to use the buffer which is on player.

File: src/main.c [hunk 4]
```diff
@@ -544,6 +740,10 @@ int _gen_sign_buffer(
     return count;
 }
 
+/**
+Creates the buffer that will be used for a sign in a chunk.
+\param[out] chunk: The chunk that contains the sign.
```
Comment (lukemoore495) [inline src/main.c]: I think this is param[in,out] as well. The chunk is passed in and modified.

File: src/main.c [hunk 5]
```diff
@@ -873,6 +1095,11 @@ void compute_chunk(WorkerItem *item)
     item->data = data;
 }
 
+/**
+Generates the chunk to prepare it to be rendered.
+\param[out] chunk: The pointer for the chunk that will be generated.
```
Comment (lukemoore495) [inline src/main.c]: I think all the chunk's being modified in these functions should be param[in,out]

**Review by lukemoore495 — CHANGES_REQUESTED** [review]
This is a significant amount of documentation. It definitely helps make sense of main.c. Nice work.

**Comment by fosterj14** [comment]
Just pushed an update that addressed your comments. Thanks for the feedback

**Review by lukemoore495 — APPROVED** [review]

**Pull request merged**